alot/crypto.py

```diff
@@ -202,12 +202,15 @@ def decrypt_verify(encrypted):
     """
     ctx = gpg.core.Context()
     try:
-        (plaintext, _, verify_result) = ctx.decrypt(encrypted, verify=True)
+        plaintext, _, verify_result = ctx.decrypt(encrypted, verify=True)
+        sigs = verify_result.signatures
     except gpg.errors.GPGMEError as e:
         raise GPGProblem(str(e), code=e.getcode())
-    # what if the signature is bad?
+    except gpg.errors.BadSignatures as e:
+        plaintext, _, _ = ctx.decrypt(encrypted, verify=False)
```

Comment:
should we catch the same GPGMEError as in line 207 for this?

Reply:
I thought about it, and I don't think we need to. I *think* that GPGMEError will be raised before BadSignatures if the content is invalid. But I'm not completely sure. That's one of the reasons I think tests are needed.